Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -2481,16 +2481,25 @@ bool CLuaPedDefs::SetPedExitVehicle(CClientPed* pPed)
     return pPed->ExitVehicle();
 }
 
-bool CLuaPedDefs::killPedTask(CClientPed* ped, taskType taskType, std::uint8_t taskNumber, std::optional<bool> gracefully) noexcept
+bool CLuaPedDefs::killPedTask(CClientPed* ped, taskType taskType, std::uint8_t taskNumber, std::optional<bool> gracefully) 
 {
     switch (taskType)
     {
         case taskType::PRIMARY_TASK:
         {
+            if (taskNumber == 4)
+                throw LuaFunctionError("Killing TASK_PRIORITY_DEFAULT is not allowed");
+
+            if (taskNumber > 3)
```

Comment:
Can we make use of the existing `enums` instead of using these 'magic' numbers?

https://github.com/multitheftauto/mtasa-blue/blob/master/Client/sdk/game/CTaskManager.h#L16-L35